test/index.js

```diff
@@ -1463,6 +1463,139 @@ describe('Nightmare', function () {
       didSubscribe.should.be.false;
     });
 
+    it('should subscribe to frames when requested necessary', function(done) {
+      var didSubscribe = false;
+      var didUnsubscribe = false;
+      var FrameManager = require('../lib/frame-manager.js');
+      var fn;
+      var manager = FrameManager({
+        webContents: {
+          debugger: {
+            isAttached: function() { return true; },
+            sendCommand: function(command) { if (command === 'DOM.highlightRect') { fn('mock-data'); }}
+          },
+          beginFrameSubscription: function(_fn) { didSubscribe = true; fn = _fn; },
+          endFrameSubscription: function() { didUnsubscribe = true; },
+          executeJavaScript: function() {}
+        }
+      });
```

Comment:
> This file is quite lengthy and isn't broken down, where would the class definition live?

Yeah, I am also reminded of how painfully long the test file is whenever I look at it—I think it’s a consequence of it having grown pretty organically and the Segment team not seeming to put much effort into this project anymore.

In the past, I’ve added support items to the bottom where there are a few such things (see [this custom Chai assertion, for example](https://github.com/segmentio/nightmare/blob/master/test/index.js#L2368-L2378)), but I think a mock class like I’m suggesting would be too big for that. I’d either:
- Just put it in a separate file in the same directory (e.g. how `server.js` is)
- Create a `test/support` directory and put it in there